src/System.Diagnostics.Process/src/System/Diagnostics/ProcessStartInfo.Unix.cs

```diff
@@ -13,26 +13,26 @@ public sealed partial class ProcessStartInfo
 
         public string UserName
         {
-            get { throw new PlatformNotSupportedException(); }
-            set { throw new PlatformNotSupportedException(); }
+            get { throw new PlatformNotSupportedException(SR.ProcessStartIdentityNotSupported); } // Do we expect to map this to sudo in due course?
```

Comment:
I don't know of any way to implement this programmatically, in particular around handling of providing the password without a password prompt.  If there is a way, we should implement it.